Clarify the exact bounds of what is required for slices (#138)
data-apis/array-api: spec/API_specification/indexing.md

```{note}

> This specification does not require "clipping" out-of-bounds slice indices (for example, in Python slice semantics, `0:100` and `0:10` are equivalent on a list of length `10`).

Reviewer: ```suggestion
This specification does not require "clipping" out-of-bounds slice indices.

```{note}

This is in contrast to Python slice semantics where `0:100` and `0:10` are equivalent on a list of length `10`.
```
```

> As such, only the following ranges for the start and stop values of a slice are required. Let `n` be the axis (dimension) size being sliced. For a slice `i:j:k`, the behavior specified above should be implemented for the following:

Reviewer: ```suggestion
The following ranges for the start and stop values of a slice must be supported. Let `n` be the axis (dimension) size being sliced. For a slice `i:j:k`, the behavior specified above should be implemented for the following:
```
